src/pages/api/umamiService.ts

```
@@ -0,0 +1,165 @@
+interface TrendingPostResponse {
+  message?: string;
+  posts?: PageView[];
+}
+
+/**
+ * Validate date format.
+ * @param date - Date string to validate.
+ * @returns {boolean} - True if the date is valid, false otherwise.
+ */
+function isValidDate(date: string): boolean {
+  return !isNaN(Date.parse(date));
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)